tool/grind.dart

```diff
@@ -274,3 +276,50 @@ Future _uploadStats(int sdkDocsGenTime) async {
     return librato.createAnnotation('build_ui', annotation);
   }
 }
+
+// TODO: check http links, check links in <link>
+@Task('Check links')
+checkLinks() {
+  bool foundError = false;
+  Set<String> visited = new Set();
+  final origin = 'test_package/doc/api/';
+  var start = 'index.html';
+
+  doCheck(String pathToCheck, [String source]) {
+    var fullPath = path.normalize("$origin$pathToCheck");
+    if (visited.contains(fullPath)) return;
+    visited.add(fullPath);
+    //print("Visiting $fullPath");
+
+    File file = new File("$fullPath");
+    if (!file.existsSync()) {
+      foundError = true;
+      print('  * Not found: $fullPath from $source');
+      return;
+    }
+    Document doc = parse(file.readAsStringSync());
+    Element base = doc.querySelector('base');
+    String baseHref;
+    if (base != null) {
+      baseHref = base.attributes['href'];
+    }
+    //print("  Base is $baseHref");
+    List<Element> links = doc.querySelectorAll('a');
+    links
+        .map((link) => link.attributes['href'])
+        .where((href) => href != null)
+        .forEach((href) {
+      if (!href.startsWith('http') && !href.contains('#')) {
+        //print("  Found link: $href");
+        var full = '${path.dirname(pathToCheck)}/$baseHref/$href';
+        var normalized = path.normalize(full);
+        //print("    => $full\n      => $normalized");
+        doCheck(normalized, pathToCheck);
+      }
+    });
+  }
+
+  doCheck(start);
+
+  if (foundError) exit(1);
```

Comment:
Dumping a whole exception here is probably not that useful for the user. Did we ever add a failWithNoException ?